File: services/metadata-service/GraNAS.Metadata.Models/Permission.cs
namespace GraNAS.Metadata.Models;

public class Permission
{
  public Guid Id { get; set; }
  public Guid FolderId { get; set; }
  public Guid UserId { get; set; }
  public AccessLevel AccessLevel { get; set; }
  public string? Path { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime? UpdatedAt { get; set; }

  public Folder? Folder { get; set; }
}

File: clients/web/src/types/permission.ts
import type { AccessLevel } from './folder';

export interface PermissionResponse {
  userId: string;
  accessLevel: AccessLevel;
  path: string | null;
  createdAt: string;
}

export interface GrantPermissionRequest {
  email: string;
  accessLevel: AccessLevel;
  path?: string | null;
}
